### DISCUSSION TITLE
How to download all videos of a youtube channel for jellyfin, using a video format suitable for browser playback

### DISCUSSION THREAD

**Discussion by FelixSchwarz** [jmbannon/ytdl-sub#989] [Q&A] [answered]
I'm trying to get `ytdl-sub` to download all videos from a certain youtube channel. The idea is that the downloaded files can be streamed with Jellyfin. My Jellyfin box has a pretty weak CPU so transcoding is not really an option.

However I can not figure out how to create a suitable configuration for that use case.

```
__preset__:
  overrides:
    tv_show_directory: "/path/downloads"


Jellyfin TV Show by Date:

  # Sets genre tag to "Online_Course"
  = Online_Course:
    "PyTorch Official": "https://www.youtube.com/@PyTorch"

```

In the docs I can see that there is a `max_1080p` preset which might be a good fit for the "video format" need but I don't know how to apply that.

In general it seems like the config yaml can contain multiple, completely different format types with a lot of magic. `config.yaml` and `subscriptions.yaml` like [this example](https://github.com/jmbannon/ytdl-sub/discussions/396) seem to be pretty straight-forward but do not provide the necessary metadata for Jellyfin.

On the other hand there are [prebuilt presets for TV Shows](https://ytdl-sub.readthedocs.io/en/latest/prebuilt_presets/tv_shows.html) which seem to work differently. Somehow the key "Plex TV Show by Date" selects a preset?

Probably everything explained in the docs but I'm struggling to piece everything together.

**Comment by jmbannon** [comment, marked as answer]
To apply multiple presets in the same key, do `Jellyfin TV Show by Date | max_1080p:`

**Reply by FelixSchwarz** [reply]
Thanks a lot, that really helped.

I have to say that the config format does not resemble any other tool I used in the past - lots of "magic" which was not obvious to me.

**Reply by jmbannon** [reply]
Yes, I need to rethink the docs to try and explain the magic better

**Comment by deedaleet** [comment]
This is hugely helpful, thank you.
The syntax of:

Jellyfin TV Show by Date | max_1080p:

or in my case what I wanted:
Plex TV Show by Date | best_video_quality:

This seems to appear nowhere in the docs or examples at all, and it took quite some time to find this solution (in my case default download behavior was maxing out at 1080p, and I wanted the available 4k).
Please consider adding this to the docs asap since setting max quality value and/or best quality seems like something nearly all users will want to do.

And thanks for this great project!
